In [2]:
import math
import pandas as pd
import scipy.stats as st

1.	Suppose a marketing research firm would like to use the movie’s budget (BUDGET in thousands dollar) as the independent variable in order to predict the U.S. box office revenue (REVENUE in thousands dollar).   35 movies are randomly selected.

In [3]:
# a - find and interpret the regression line

sm_x = 2957.5
sm_xx = 23775.0417
sm_y = 4112.1906
# sm_yy = 127415.4
sm_xy = 47093.6722
n = 35
df = n - 2

y_bar = sm_y / n
x_bar = sm_x / n

b1 = (sm_xy - ((sm_x * sm_y) / n)) / (sm_xx - (sm_x**2 / n))

b1 = round(b1, 4)
print(f"coefficient: {b1}")

b0 = round(y_bar - (b1 * x_bar), 4)
print(f"intercept: {b0}")

print(f"Linear equation: revenue in thousands = {b0} + ({b1} * (budget))")

coefficient: 1.3284
intercept: 5.2414
Linear equation: revenue in thousands = 5.2414 + (1.3284 * (budget))


When establishing a movie's budget, the general revenue can be predicted - though it ignores rating. When establishing a budget, your estimated revenue begins around $5200, and multiplies by 1.3284 for every dollar you put in, assuming a profit given your movie performs and rates well.

In [4]:
# b - Using the regression line found in part a), find the estimated U.S. box office revenue when the movie’s 
#     budget is 100 thousands dollar.

budget = 100  # in thousands of dollars

pred = round(b0 + (b1 * budget), 2)

print(f"Assuming a budget of ${budget * 1000}, our estimated revenue is: ${pred * 1000}")

Assuming a budget of $100000, our estimated revenue is: $138080.0


In [5]:
# c - Find 95% prediction interval for part b

s2_xy = 9770.228
alpha = 0.05 / 2

crit_val = abs(st.t.ppf(alpha, df))

prod = round(crit_val * math.sqrt(s2_xy * (1 + (1 / n) + ( (pred - x_bar)**2) / sm_xx )), 4)
print(f"Product: {prod}")

lower = round(pred - prod, 4)
upper = pred + prod

print(f"{(1 - (alpha * 2))*100}% prediction interval: ({lower}, {upper})")

Product: 215.5926
95.0% prediction interval: (-77.5126, 353.6726)


This is a helpful way to understand the volatility of developing film. We do not have negative revenue, that is not possible.

2.	Below is part of the output from another 35 movies done by excel

                Coefficient     Standard Error  t stat  P-value
        Intercept   32.0943	    60.168	0.533	0.5983
        BUDGET	    -0.4214	    0.3034	-1.389	0.1767

In [6]:
# 2 - Find a 90% confidence interval for the slope of budget

budget = b1 = -0.4214
n = 35
s_err = 0.3034
df = n - 2

alpha = 0.10 / 2

crit_val = st.t.ppf(1 - alpha, df)

prod = crit_val * s_err

lower = round(budget - prod, 4)
upper = round(budget + prod, 4)

print(f"{(1 - (alpha * 2))*100}% confidence interval: ({lower}, {upper})")

90.0% confidence interval: (-0.9349, 0.0921)


This is an interesting line to interpret, it feels as though most movies result in losses.

3.	In a related study, the marketing research firm would like to predict total U.S. box office revenue. 35 movies are randomly selected, and four measurements are taken for each movie: movie’s budget (BUDGET, in thousands dollar), opening-weekend revenue (OPENING, in thousands dollar), opening-weekend theater count (THEATERS) and the U.S. box office revenue (REVENUE, in thousands dollar).        

In [10]:
# a - complete a linear ANOVA table

n = 35
vals = 4

ssr = 330786.1
sst = 366849.7

sse = sst - ssr

print(f"SSR: {ssr}")
print(f"SSE: {sse}")
print(f"SST: {sst}")

df_reg = vals - 1
df_err = n - vals
df_tot = n - 1

print(f"df reg: {df_reg}")
print(f"df err: {df_err}")
print(f"df tot: {df_tot}")

msr = ssr / df_reg
mse = sse / df_err

print(f"MSR: {msr}")
print(f"MSE: {mse}")

f = msr / mse
print(f"F-val: {f}")

p_val = st.f.cdf(f, df_reg, df_err)
print(f"P val: {1 - p_val}")

SSR: 330786.1
SSE: 36063.600000000035
SST: 366849.7
df reg: 3
df err: 31
df tot: 34
MSR: 110262.03333333333
MSE: 1163.341935483872
F-val: 94.78041663431632
P val: 9.992007221626409e-16


### ANOVA table

    Source          SS          df      MS          F
        Regression  330786.1    3       110262.033  94.78
        Error       36063.6     31      1163.341
        Total       366849.7    34      

In [11]:
r2 = ssr / sst
print(f"R^2: {r2}")

R^2: 0.9016937999404114


Our model is able to explain ~ 90% of variation as it pertains to our prediction interval. 